x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.tsx

```diff
@@ -301,28 +301,68 @@ const AlertsTableStateWithQueryProvider = memo(
       initialBrowserFields: propBrowserFields,
     });
 
-    const [
-      isLoading,
-      {
-        alerts,
-        oldAlertsData,
-        ecsAlertsData,
-        isInitializing,
-        getInspectQuery,
-        refetch: refresh,
-        totalAlerts: alertsCount,
-      },
-    ] = useFetchAlerts({
-      fields,
+    const [queryParams, setQueryParams] = useState({
       featureIds,
+      fields,
       query,
-      pagination,
-      onPageChange,
-      onLoaded,
-      runtimeMappings,
       sort,
-      skip: false,
+      runtimeMappings,
+      pageIndex: 0,
+      pageSize: initialPageSize,
+    });
+
+    useEffect(() => {
```

Comment:
Let's apply the side effects on event handlers instead of useEffect

Reply:
I agree with your point and would have liked to avoid a `useEffect` however here the changes don't come from an event handler but from the component props themselves. Also, the reason why this intermediate state exists is I have to batch the state updates into one to reset the pagination correctly when other query params change, otherwise I get an unwanted request with the wrong pagination parameters. I could use a `useMemo`, but I'm afraid other useEffects would be required to keep references to the previous values of the query params...

Reply:
alright not a big deal then